### PR TITLE
expr: Implemented __trunc__ method

### PR DESCRIPTION
**Pull request by abaokar-07**
Added the __trunc__ method as the one in math.trunc method of the math module

@asmeurer please review

### CODE REVIEW

File: sympy/core/expr.py [hunk 1]
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
                 return sympify(dif.is_negative)
         return StrictLessThan(self, other, evaluate=False)
 
+    def __trunc__(self):
+        r = self.round(2)
+        if self in(S.Nan, S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
```
Comment (asmeurer) [inline sympy/core/expr.py]: Put a space after `in`

File: sympy/core/expr.py [hunk 2]
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
                 return sympify(dif.is_negative)
         return StrictLessThan(self, other, evaluate=False)
 
+    def __trunc__(self):
+        r = self.round(2)
```
Comment (asmeurer) [inline sympy/core/expr.py]: Why not just use `int(self)` here?

File: sympy/core/expr.py [hunk 3]
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
...
+        if self in(S.Nan, S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
+            raise TypeError("can't truncate", self)
+        if not r.is_Number:
+            raise TypeError("can't truncate symbols and expressions")
+        else:
+            return int(r)
```
Comment (asmeurer) [inline sympy/core/expr.py]: This should return a SymPy type.

**Comment by asmeurer** [comment]
Please add tests.

**Comment by abaokar-07** [comment]
@asmeurer I have modified `expr.py` according to your reviews
But before adding tests, I wanted to ask one thing that after I modified expr, I gave input as `trunc(3.67)` and I got the following error : `TypeError: trunc() missing 1 required positional argument: 'p'`

**Comment by asmeurer** [comment]
You need to use `math.trunc`. The `trunc` that comes with SymPy is unrelated.

**Comment by abaokar-07** [comment]
@asmeurer Added the tests

**Comment by abaokar-07** [comment]
@asmeurer what's the status on this, any more changes are needed or is it fine?

File: sympy/core/expr.py [hunk 4]
```diff
@@ -344,6 +344,15 @@ def __lt__(self, other):
                 return sympify(dif.is_negative)
         return StrictLessThan(self, other, evaluate=False)
 
+    def __trunc__(self):
+        if not self.is_number:
+            raise TypeError("can't truncate symbols and expressions")
+        r = int(self)
+        if r in (S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
```
Comment (asmeurer) [inline sympy/core/expr.py]: r cannot be one of these.

**Comment by abaokar-07** [comment]
@asmeurer are the latest changes fine?

**Comment by abaokar-07** [comment]
@asmeurer is the modified form okay?

**Comment by abaokar-07** [comment]
@asmeurer ping please, is the change good?

File: sympy/core/expr.py [hunk 5]
```diff
@@ -344,6 +344,14 @@ def __lt__(self, other):
                 return sympify(dif.is_negative)
         return StrictLessThan(self, other, evaluate=False)
 
+    def __trunc__(self):
+        if not self.is_number:
+            raise TypeError("can't truncate symbols and expressions")
+        elif self in (S.NegativeInfinity, S.Infinity, S.ComplexInfinity):
```
Comment (asmeurer) [inline sympy/core/expr.py]: I don't think this block is needed at all. This already raises TypeError.

**Comment by abaokar-07** [comment]
@asmeurer removed the code for complex, also the tests have passed

**Comment by abaokar-07** [comment]
@asmeurer are the lastest changes up to the mark?

**Comment by abaokar-07** [comment]
@asmeurer thanks for suggestions!

**Pull request merged**